guide/secure_secret_management.md

```diff
@@ -0,0 +1,165 @@
+# Guide on using Docker Swarm Secrets with OpenFaaS
+
+OpenFaaS deploys functions as Docker Swarm Services, as result there are several features that we can leverage to simplify the development and subsquent deployment of functions to hardened production environments.
+
+## Using Environment Variables
+First, and most simple, is the ability to set environment variables at deploy time. For example, you might want to set the `NODE_ENV` or `DEBUG` variable.  Setting the `NODE_ENV` while using the `faas-cli` might look like this
+
+```sh
+$ faas-cli deploy -f ./samples.yml
+$ fass-cli invoke nodehelloenv
+Hello from a production machine
+```
+
+Where your `samples.yml` stack file looks like this
+```yaml
+provider:
+  name: faas
+  gateway: http://localhost:8080
+
+functions:
+  nodehelloenv:
+    lang: Dockerfile
+    skip_build: true
+    image: functions/nodehelloenv:latest
+    environment:
+        NODE_ENV: production
+```
+
+
+(_Optional_) If you are directly using the OpenFaaS Gateway API, then it will look like this
```

Comment:
I had thought we were pushing people towards the CLI rather than providing curl examples - @alexellis wdyt?

Reply:
I can remove them completely, keep the guide simple.  I have been directly using the api a lot, so it is where my first examples come from. The cli doesn't always expose all of the features of the gateway right away (or at least this was true in the past)